### ISSUE TITLE
Strange result for the eval() function

### ISSUE THREAD

**Issue by zhjsun** [dacelib/dace#13]
I'm trying to evaluate a weighted-order function. For example:
```
#include <dace/dace.h>
#include <cmath>
#include <iostream>
#include <iomanip>

using namespace std;
using namespace DACE;

int main( void )
{
    
    DA::init( 20, 2 );

    // the first case
    DA x1 = DA(1) * DA(1) * DA(1);
    DA y1 = DA(2);
    DA z1 = exp(x1) * exp(y1);

    vector<DA> p1({DA(1), 0});
    DA f1 = z1.eval(p1);
    
    // the second case
    DA x2 = DA(1);
    DA y2 = DA(2) * DA(2) * DA(2);
    DA z2 = exp(x2) * exp(y2);

    vector<DA> p2({0, DA(2)});
    DA f2 = z2.eval(p2);

    cout << "z1 function" << endl << z1 << endl;
    cout << "z2 function" << endl << z2 << endl;
    cout << "f1 function" << endl << f1 << endl;
    cout << "f2 function" << endl << f2 << endl;

    return 0;
}
```
The result of f1 and f2 are supposed the same in some sense. However, it seems that some higher-order terms are missed in f2. The results are as follows.
```
z1 function
     I  COEFFICIENT              ORDER EXPONENTS
     1    1.0000000000000000e+00   0   0  0
     2    1.0000000000000000e+00   1   0  1
     3    5.0000000000000000e-01   2   0  2
     4    1.6666666666666666e-01   3   0  3
     5    1.0000000000000000e+00   3   3  0
     6    4.1666666666666664e-02   4   0  4
     7    1.0000000000000000e+00   4   3  1
     8    8.3333333333333332e-03   5   0  5
     9    5.0000000000000000e-01   5   3  2
    10    1.3888888888888889e-03   6   0  6
    11    1.6666666666666666e-01   6   3  3
    12    5.0000000000000000e-01   6   6  0
    13    1.9841269841269841e-04   7   0  7
    14    4.1666666666666664e-02   7   3  4
    15    5.0000000000000000e-01   7   6  1
    16    2.4801587301587302e-05   8   0  8
    17    8.3333333333333332e-03   8   3  5
    18    2.5000000000000000e-01   8   6  2
    19    2.7557319223985893e-06   9   0  9
    20    1.3888888888888889e-03   9   3  6
    21    8.3333333333333329e-02   9   6  3
    22    1.6666666666666666e-01   9   9  0
    23    2.7557319223985894e-07  10   0 10
    24    1.9841269841269841e-04  10   3  7
    25    2.0833333333333332e-02  10   6  4
    26    1.6666666666666666e-01  10   9  1
    27    2.5052108385441720e-08  11   0 11
    28    2.4801587301587302e-05  11   3  8
    29    4.1666666666666666e-03  11   6  5
    30    8.3333333333333329e-02  11   9  2
    31    2.0876756987868100e-09  12   0 12
    32    2.7557319223985893e-06  12   3  9
    33    6.9444444444444447e-04  12   6  6
    34    2.7777777777777776e-02  12   9  3
    35    4.1666666666666664e-02  12  12  0
    36    1.6059043836821616e-10  13   0 13
    37    2.7557319223985894e-07  13   3 10
    38    9.9206349206349206e-05  13   6  7
    39    6.9444444444444441e-03  13   9  4
    40    4.1666666666666664e-02  13  12  1
    41    1.1470745597729726e-11  14   0 14
    42    2.5052108385441720e-08  14   3 11
    43    1.2400793650793651e-05  14   6  8
    44    1.3888888888888887e-03  14   9  5
    45    2.0833333333333332e-02  14  12  2
    46    7.6471637318198174e-13  15   0 15
    47    2.0876756987868100e-09  15   3 12
    48    1.3778659611992946e-06  15   6  9
    49    2.3148148148148149e-04  15   9  6
    50    6.9444444444444441e-03  15  12  3
    51    8.3333333333333332e-03  15  15  0
    52    4.7794773323873859e-14  16   0 16
    53    1.6059043836821616e-10  16   3 13
    54    1.3778659611992947e-07  16   6 10
    55    3.3068783068783064e-05  16   9  7
    56    1.7361111111111110e-03  16  12  4
    57    8.3333333333333332e-03  16  15  1
    58    2.8114572543455210e-15  17   0 17
    59    1.1470745597729726e-11  17   3 14
    60    1.2526054192720860e-08  17   6 11
    61    4.1335978835978830e-06  17   9  8
    62    3.4722222222222218e-04  17  12  5
    63    4.1666666666666666e-03  17  15  2
    64    1.5619206968586228e-16  18   0 18
    65    7.6471637318198174e-13  18   3 15
    66    1.0438378493934050e-09  18   6 12
    67    4.5928865373309819e-07  18   9  9
    68    5.7870370370370373e-05  18  12  6
    69    1.3888888888888887e-03  18  15  3
    70    1.3888888888888889e-03  18  18  0
    71    8.2206352466243310e-18  19   0 19
    72    4.7794773323873859e-14  19   3 16
    73    8.0295219184108080e-11  19   6 13
    74    4.5928865373309820e-08  19   9 10
    75    8.2671957671957661e-06  19  12  7
    76    3.4722222222222218e-04  19  15  4
    77    1.3888888888888889e-03  19  18  1
    78    4.1103176233121653e-19  20   0 20
    79    2.8114572543455210e-15  20   3 17
    80    5.7353727988648631e-12  20   6 14
    81    4.1753513975736200e-09  20   9 11
    82    1.0333994708994708e-06  20  12  8
    83    6.9444444444444444e-05  20  15  5
    84    6.9444444444444447e-04  20  18  2
------------------------------------------------

z2 function
     I  COEFFICIENT              ORDER EXPONENTS
     1    1.0000000000000000e+00   0   0  0
     2    1.0000000000000000e+00   1   1  0
     3    5.0000000000000000e-01   2   2  0
     4    1.0000000000000000e+00   3   0  3
     5    1.6666666666666666e-01   3   3  0
     6    1.0000000000000000e+00   4   1  3
     7    4.1666666666666664e-02   4   4  0
     8    5.0000000000000000e-01   5   2  3
     9    8.3333333333333332e-03   5   5  0
    10    5.0000000000000000e-01   6   0  6
    11    1.6666666666666666e-01   6   3  3
    12    1.3888888888888889e-03   6   6  0
    13    5.0000000000000000e-01   7   1  6
    14    4.1666666666666664e-02   7   4  3
    15    1.9841269841269841e-04   7   7  0
    16    2.5000000000000000e-01   8   2  6
    17    8.3333333333333332e-03   8   5  3
    18    2.4801587301587302e-05   8   8  0
    19    1.6666666666666666e-01   9   0  9
    20    8.3333333333333329e-02   9   3  6
    21    1.3888888888888889e-03   9   6  3
    22    2.7557319223985893e-06   9   9  0
    23    1.6666666666666666e-01  10   1  9
    24    2.0833333333333332e-02  10   4  6
    25    1.9841269841269841e-04  10   7  3
    26    2.7557319223985894e-07  10  10  0
    27    8.3333333333333329e-02  11   2  9
    28    4.1666666666666666e-03  11   5  6
    29    2.4801587301587302e-05  11   8  3
    30    2.5052108385441720e-08  11  11  0
    31    4.1666666666666664e-02  12   0 12
    32    2.7777777777777776e-02  12   3  9
    33    6.9444444444444447e-04  12   6  6
    34    2.7557319223985893e-06  12   9  3
    35    2.0876756987868100e-09  12  12  0
    36    4.1666666666666664e-02  13   1 12
    37    6.9444444444444441e-03  13   4  9
    38    9.9206349206349206e-05  13   7  6
    39    2.7557319223985894e-07  13  10  3
    40    1.6059043836821616e-10  13  13  0
    41    2.0833333333333332e-02  14   2 12
    42    1.3888888888888887e-03  14   5  9
    43    1.2400793650793651e-05  14   8  6
    44    2.5052108385441720e-08  14  11  3
    45    1.1470745597729726e-11  14  14  0
    46    8.3333333333333332e-03  15   0 15
    47    6.9444444444444441e-03  15   3 12
    48    2.3148148148148149e-04  15   6  9
    49    1.3778659611992946e-06  15   9  6
    50    2.0876756987868100e-09  15  12  3
    51    7.6471637318198174e-13  15  15  0
    52    8.3333333333333332e-03  16   1 15
    53    1.7361111111111110e-03  16   4 12
    54    3.3068783068783064e-05  16   7  9
    55    1.3778659611992947e-07  16  10  6
    56    1.6059043836821616e-10  16  13  3
    57    4.7794773323873859e-14  16  16  0
    58    4.1666666666666666e-03  17   2 15
    59    3.4722222222222218e-04  17   5 12
    60    4.1335978835978830e-06  17   8  9
    61    1.2526054192720860e-08  17  11  6
    62    1.1470745597729726e-11  17  14  3
    63    2.8114572543455210e-15  17  17  0
    64    1.3888888888888889e-03  18   0 18
    65    1.3888888888888887e-03  18   3 15
    66    5.7870370370370373e-05  18   6 12
    67    4.5928865373309819e-07  18   9  9
    68    1.0438378493934050e-09  18  12  6
    69    7.6471637318198174e-13  18  15  3
    70    1.5619206968586228e-16  18  18  0
    71    1.3888888888888889e-03  19   1 18
    72    3.4722222222222218e-04  19   4 15
    73    8.2671957671957661e-06  19   7 12
    74    4.5928865373309820e-08  19  10  9
    75    8.0295219184108080e-11  19  13  6
    76    4.7794773323873859e-14  19  16  3
    77    8.2206352466243310e-18  19  19  0
    78    6.9444444444444447e-04  20   2 18
    79    6.9444444444444444e-05  20   5 15
    80    1.0333994708994708e-06  20   8 12
    81    4.1753513975736200e-09  20  11  9
    82    5.7353727988648631e-12  20  14  6
    83    2.8114572543455210e-15  20  17  3
    84    4.1103176233121653e-19  20  20  0
------------------------------------------------

f1 function
     I  COEFFICIENT              ORDER EXPONENTS
     1    1.0000000000000000e+00   0   0  0
     2    1.0000000000000000e+00   3   3  0
     3    5.0000000000000000e-01   6   6  0
     4    1.6666666666666666e-01   9   9  0
     5    4.1666666666666664e-02  12  12  0
     6    8.3333333333333332e-03  15  15  0
     7    1.3888888888888889e-03  18  18  0
------------------------------------------------

f2 function
     I  COEFFICIENT              ORDER EXPONENTS
     1    1.0000000000000000e+00   0   0  0
     2    1.0000000000000000e+00   3   0  3
------------------------------------------------
```

**Comment by abgandar** [comment]
Thanks for bringing this problem to our attention. It seems something is going wrong deep inside the evaluation code when multiple variables are involved. I'm investigating this issue right now. This is likely a relatively new bug, as the evaluation was definitely working in earlier releases...

**Comment by abgandar** [comment]
Closed with fix in pull request #14 .
We will release an updated version of the DACE binaries shortly. In the meantime, please update to the latest code and rebuild the DACE library locally.
